scripts/optimize_power_system.py

```diff
@@ -0,0 +1,22 @@
+import core
+import dto
+import io
+import plotting_utils
+
+
+# Read in data, validate date et.c. with methods from io
+some_test_data_file = 'test_data.json'
+json_data = io.file_io.read_data_from_json(some_test_data_file)
+
+# deserialize json data to objects
+power_network: dto.PowerNetwork = dto.deserialize_power_network_data(json_data)
+
+# call optimization method, input are data objects
+optimized_result = core.optimize_power_system(nodes=power_network.nodes, edges=power_network.edges, constraints=power_network.constraints)
```

Comment:
This is just an example to show how the structure is used to divide different parts of the "work", io (file reading et.c.) is one separate thing, validation of the inputdata another, then the data may be deserialized to data objects which is also a separate step, then the actual optimization which is the "core" of this repo, what we really want to focus on. In the end there might be some serialization, output to file, plotting et.c.

The main product will be in core. In this example just a method to show the separation of core from the rest, but might as well be one or several classes with methods.